pennylane/transforms/broadcast_expand.py

```diff
@@ -45,6 +45,11 @@ def _split_operations(ops, num_tapes):
     return new_ops
 
 
+# def _unwrap_counts(res):
+#     """Helper function to unwrap counts dictionaries from 0-D arrays."""
+#     return [r.item() for r in res]
```

Comment:
Has there been discussion already on whether we want to do this automatically? returning these dicts wrapped in an array surely sounds suboptimal :sweat_smile: Would be fine by me to postpone this to a small follow-up PR, if it is tracked somewhere. Then these lines should be removed.

Reply:
Actually, this is one of the open questions for this PR. I would say that this is a very low priority change because the wrapping only occurs when the batch size is 1. Would appreciate feedback on whether we should tackle this now or leave it for later. cc @trbromley

Reply:
Fine by me either way!